medusa/search/proper.py

```diff
@@ -328,43 +329,59 @@ def _download_propers(self, proper_list):
+                    ),
+                    amount=len(candidate.actual_episodes) * 2,
+                ),
+                [candidate.indexerid, candidate.actual_season, candidate.quality,
                  history_limit.strftime(History.date_format),
                  DOWNLOADED, SNATCHED, SNATCHED_PROPER, SNATCHED_BEST])
 
-            # make sure that none of the existing history downloads are the same proper we're trying to download
-            # if the result exists in history already we need to skip it
-            clean_proper_name = self._canonical_name(cur_proper.name, clear_extension=True)
-            if any(clean_proper_name == self._canonical_name(cur_result[b'resource'], clear_extension=True)
-                   for cur_result in history_results):
-                log.debug('This proper {result!r} is already in history, skipping it', {'result': cur_proper.name})
-                continue
-            else:
-                # make sure that none of the existing history downloads are the same proper we're trying to download
-                clean_proper_name = self._canonical_name(cur_proper.name)
-                if any(clean_proper_name == self._canonical_name(cur_result[b'resource'])
-                       for cur_result in history_results):
-                    log.debug('This proper {result!r} is already in history, skipping it', {'result': cur_proper.name})
-                    continue
-
-                cur_proper.create_episode_object()
+            proper_tags_len = len(candidate.proper_tags)
+            proper_name = self._canonical_name(candidate.name, clear_extension=True)
+            proper_name_ext = self._canonical_name(candidate.name)
+
+            for result in history_results:
+
+                proper_tags = result[b'proper_tags']
+                if proper_tags and len(proper_tags.split('|')) >= proper_tags_len:
+                    log.debug(
+                        'Current release has the same or more proper tags,'
+                        ' skipping new proper {result!r}',
+                        {'result': candidate.name},
+                    )
+                    break
+
+                # make sure that none of the existing history downloads are the same proper we're
+                # trying to downloadif the result exists in history already we need to skip it
+                if proper_name == self._canonical_name(
+                    result[b'resource'], clear_extension=True
+                ) or proper_name_ext == self._canonical_name(result[b'resource']):
+                    log.debug(
+                        'This proper {result!r} is already in history, skipping it',
+                        {'result': candidate.name},
+                    )
+                    break
 
+            else:
+                candidate.create_episode_object()
                 # snatch it
-                snatch_episode(cur_proper)
-                time.sleep(cpu_presets[app.CPU_PRESET])
```

Comment:
Isn't this in place to keep CPU usage low (or high), per user preference?

Reply:
No, time.sleep() just stops code execution and nothing else. It doesn't give "a break" to the CPU, more likely the contrary. This kind of things shouldn't be implemented manually in high-level programming languages.